# 1. 修复路径解析不一致的 warnings
# <w> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Resolving '../../../../css-loader@6.7.3_webpack@5.75.0/node_modules/css-loader/dist/cjs' in D:\2\ddb\ext\node_modules\.pnpm\loader-runner@4.3.0\node_modules\loader-runner\lib for build dependencies doesn't lead to expected result 'd:\2\ddb\ext\node_modules\.pnpm\css-loader@6.7.3_webpack@5.75.0\node_modules\css-loader\dist\cjs.js', but to 'D:\2\ddb\ext\node_modules\.pnpm\css-loader@6.7.3_webpack@5.75.0\node_modules\css-loader\dist\cjs.js' instead. Resolving dependencies are ignored for this path.
# <w>  at resolve commonjs file ../../../../css-loader@6.7.3_webpack@5.75.0/node_modules/css-loader/dist/cjs (expected d:\2\ddb\ext\node_modules\.pnpm\css-loader@6.7.3_webpack@5.75.0\node_modules\css-loader\dist\cjs.js)
# <w>  at file dependencies D:\2\ddb\ext\node_modules\.pnpm\loader-runner@4.3.0\node_modules\loader-runner\lib\loadLoader.js
# <w>  at file D:\2\ddb\ext\node_modules\.pnpm\loader-runner@4.3.0\node_modules\loader-runner\lib\loadLoader.js
# <w>  at file dependencies D:\2\ddb\ext\node_modules\.pnpm\loader-runner@4.3.0\node_modules\loader-runner\lib\LoaderRunner.js
# <w>  at file D:\2\ddb\ext\node_modules\.pnpm\loader-runner@4.3.0\node_modules\loader-runner\lib\LoaderRunner.js
# <w>  at file dependencies D:\2\ddb\ext\node_modules\.pnpm\webpack@5.75.0\node_modules\webpack\lib\NormalModuleFactory.js
# <w>  at file D:\2\ddb\ext\node_modules\.pnpm\webpack@5.75.0\node_modules\webpack\lib\NormalModuleFactory.js
# <w>  at file dependencies D:\2\ddb\ext\node_modules\.pnpm\webpack@5.75.0\node_modules\webpack\lib\Compiler.js
# <w>  at file D:\2\ddb\ext\node_modules\.pnpm\webpack@5.75.0\node_modules\webpack\lib\Compiler.js
# <w>  at file dependencies D:\2\ddb\ext\node_modules\.pnpm\webpack@5.75.0\node_modules\webpack\lib\webpack.js
# <w>  at file D:\2\ddb\ext\node_modules\.pnpm\webpack@5.75.0\node_modules\webpack\lib\webpack.js
# <w>  at file dependencies D:\2\ddb\ext\node_modules\.pnpm\webpack@5.75.0\node_modules\webpack\lib\index.js
# <w>  at file D:\2\ddb\ext\node_modules\.pnpm\webpack@5.75.0\node_modules\webpack\lib\index.js
# <w>  at file dependencies D:\2\ddb\ext\node_modules\.pnpm\ts-loader@9.4.2_typescript@4.9.5_webpack@5.75.0\node_modules\ts-loader\dist\utils.js
# <w>  at file D:\2\ddb\ext\node_modules\.pnpm\ts-loader@9.4.2_typescript@4.9.5_webpack@5.75.0\node_modules\ts-loader\dist\utils.js
# <w>  at file dependencies D:\2\ddb\ext\node_modules\.pnpm\ts-loader@9.4.2_typescript@4.9.5_webpack@5.75.0\node_modules\ts-loader\dist\index.js
# <w>  at file D:\2\ddb\ext\node_modules\.pnpm\ts-loader@9.4.2_typescript@4.9.5_webpack@5.75.0\node_modules\ts-loader\dist\index.js
# <w>  at file dependencies D:\2\ddb\ext\node_modules\.pnpm\ts-loader@9.4.2_typescript@4.9.5_webpack@5.75.0\node_modules\ts-loader\index.js
# <w>  at file D:\2\ddb\ext\node_modules\.pnpm\ts-loader@9.4.2_typescript@4.9.5_webpack@5.75.0\node_modules\ts-loader\index.js
# <w>  at resolve commonjs D:\2\ddb\ext\node_modules\.pnpm\ts-loader@9.4.2_typescript@4.9.5_webpack@5.75.0\node_modules\ts-loader\index.js
# 
# 
# 2. 不要打印下面这样的 warning，两个路径的区别只有盘符的大小写
# 更好的修复方法是在解析、创建 webpack 模块时就统一路径的格式
# WARNING in ./node_modules/.pnpm/css-loader@6.7.3_webpack@5.75.0/node_modules/css-loader/dist/runtime/api.js
# There are multiple modules with names that only differ in casing.
# This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
# Use equal casing. Compare these module identifiers:
# * D:\2\ddb\web\node_modules\.pnpm\source-map-loader@4.0.1_webpack@5.75.0\node_modules\source-map-loader\dist\cjs.js!D:\2\ddb\web\node_modules\.pnpm\css-loader@6.7.3_webpack@5.75.0\node_modules\css-loader\dist\runtime\api.js
#     Used by 3 module(s), i. e.
#     D:\2\ddb\web\node_modules\.pnpm\css-loader@6.7.3_webpack@5.75.0\node_modules\css-loader\dist\cjs.js??ruleSet[1].rules[2].use[1]!D:\2\ddb\web\node_modules\.pnpm\sass-loader@13.2.0_sass@1.58.0_webpack@5.75.0\node_modules\sass-loader\dist\cjs.js??ruleSet[1].rules[2].use[2]!D:\2\ddb\web\node_modules\.pnpm\xshell@1.0.7_react-dom@18.2.0\node_modules\xshell\scroll-bar.sass
# * D:\2\ddb\web\node_modules\.pnpm\source-map-loader@4.0.1_webpack@5.75.0\node_modules\source-map-loader\dist\cjs.js!d:\2\ddb\web\node_modules\.pnpm\css-loader@6.7.3_webpack@5.75.0\node_modules\css-loader\dist\runtime\api.js
#     Used by 11 module(s), i. e.
#     d:\2\ddb\web\node_modules\.pnpm\css-loader@6.7.3_webpack@5.75.0\node_modules\css-loader\dist\cjs.js??ruleSet[1].rules[2].use[1]!d:\2\ddb\web\node_modules\.pnpm\sass-loader@13.2.0_sass@1.58.0_webpack@5.75.0\node_modules\sass-loader\dist\cjs.js??ruleSet[1].rules[2].use[2]!d:\2\ddb\web\console\dashboard.sass
#  @ ./node_modules/.pnpm/css-loader@6.7.3_webpack@5.75.0/node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[2].use[1]!./node_modules/.pnpm/sass-loader@13.2.0_sass@1.58.0_webpack@5.75.0/node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[2].use[2]!./node_modules/.pnpm/xshell@1.0.7_react-dom@18.2.0/node_modules/xshell/scroll-bar.sass 3:0-127 4:30-57
#  @ ./node_modules/.pnpm/xshell@1.0.7_react-dom@18.2.0/node_modules/xshell/scroll-bar.sass 8:6-291 22:17-24 26:0-261 26:0-261 27:22-29 27:33-47 27:50-64
#  @ ./console/window.tsx 1:0-32

diff --git a/lib/FileSystemInfo.js b/lib/FileSystemInfo.js
index 87221acfa2583ef7690d79b25bed9f6b4dd8ff9a..4043921e2cb009402c4b57a7e93e7fe1f2f51e8f 100644
--- a/lib/FileSystemInfo.js
+++ b/lib/FileSystemInfo.js
@@ -1471,8 +1471,8 @@ class FileSystemInfo {
 					resolveResults.set(key, undefined);
 					resolve(context, path, resolverContext, (err, _, result) => {
 						if (typeof expected === "string") {
-							if (!err && result && result.path === expected) {
-								resolveResults.set(key, result.path);
+							if (!err && result && result.path.toLowerCase() === expected.toLowerCase()) {
+								resolveResults.set(key, expected);
 							} else {
 								invalidResolveResults.add(key);
 								this.logger.warn(
diff --git a/lib/WebpackOptionsApply.js b/lib/WebpackOptionsApply.js
index 96a485f7400a0bc49b1ab8b2b88f36f157ce7840..2651f3222084bbc901a9bf20ed1bd769a557f039 100644
--- a/lib/WebpackOptionsApply.js
+++ b/lib/WebpackOptionsApply.js
@@ -587,7 +587,7 @@ class WebpackOptionsApply extends OptionsApply {
 			portableIds: options.optimization.portableRecords
 		}).apply(compiler);
 
-		new WarnCaseSensitiveModulesPlugin().apply(compiler);
+		// new WarnCaseSensitiveModulesPlugin().apply(compiler);
 
 		const AddManagedPathsPlugin = require("./cache/AddManagedPathsPlugin");
 		new AddManagedPathsPlugin(